### PR TITLE
Fix #483 #484 If-Match when resource does not exists and If-None-Match when it exists.

### LINKED ISSUE #483: Handling of If-Match: * is not RFC compliant
The documentation says if we omit `If-Match` header for WebDAV `PUT` method, it defaults to `*`.
https://personium.io/docs/en/apiref/current/312_Register_and_Update_WebDAV.html

This is not compliant to RFC 7232:

> If the field-value is "*", the condition is false if the origin server does not have a current representation for the target resource.

https://tools.ietf.org/html/rfc7232#section-3.1

Omitting `If-Match` header and giving `If-Match` header with the value `*` must behave differently.

I have not checked actual behavior. Is this just a error of documentation, or bug of actual implementation?

### PR DESCRIPTION
**Pull request by shimono**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <groupId>io.personium</groupId>
     <artifactId>personium-core</artifactId>
     <packaging>war</packaging>
-    <version>1.7.19</version>
+    <version>1.7.20</version>
```
Comment (tochi-y) [inline pom.xml]: It should be 1.7.20-SNAPSHOT while develop.

**Comment by shimono** [comment]
should also include #484 fix, mayby

File: src/test/java/io/personium/core/rs/box/DavFileResourceTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    /**
+     * Test delete().
+     * error.
+     * recursiveHeader is unexpected value.
+     */
+    @Test
+    public void put_IfNoneMatchWildcard_ShouldFail() {
+
+        // Mock settings
+        davFileResource = spy(new DavFileResource(null, null));
```
Comment (tochi-y) [inline src/test/java/io/personium/core/rs/box/DavFileResourceTest.java]: @shimono Test target should not be a mock as even spy clones original class.

Correct:

```java
davFileResource = new DavFileResource(null, null);
````

File: src/test/java/io/personium/core/rs/box/DavFileResourceTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+        // Run method
+        try {
+        	InputStream is = new ByteArrayInputStream("{test:1}".getBytes(Charsets.UTF_8));
```
Comment (tochi-y) [inline src/test/java/io/personium/core/rs/box/DavFileResourceTest.java]: There is a tab.

Comment (shimono) [reply src/test/java/io/personium/core/rs/box/DavFileResourceTest.java]: Fixed. Thanks for pointing it out.

File: src/test/java/io/personium/core/rs/box/DavFileResourceTest.java (hunk 2 above)
Comment (shimono) [reply src/test/java/io/personium/core/rs/box/DavFileResourceTest.java]: Fixed. Thanks for pointing it out.

**Pull request merged**